packages/ember-toucan-form/src/-private/textarea-field.hbs

```diff
@@ -18,66 +18,174 @@
+        @hint={{@hint}}
+        @error={{this.mapErrors field.rawErrors}}
+        @value={{this.assertString field.value}}
+        {{! @glint-expect-error }}
+        @onChange={{field.setValue}}
+        @isDisabled={{@isDisabled}}
+        @isReadOnly={{@isReadOnly}}
+        @rootTestSelector={{@rootTestSelector}}
+        name={{@name}}
+        ...attributes
+      >
+        <:label>{{yield to='label'}}</:label>
+        <:secondary as |secondary|>
+          {{yield
+            (hash
+              CharacterCount=(component
+                (ensure-safe-component secondary.CharacterCount)
+              )
+            )
+            to='secondary'
+          }}
+        </:secondary>
+      </Form::Fields::Textarea>
+    {{else if
+      (this.hasHintAndLabelBlocks (has-block 'label') (has-block 'hint'))
+    }}
+      <Form::Fields::Textarea
+        @error={{this.mapErrors field.rawErrors}}
+        @value={{this.assertString field.value}}
+        {{! @glint-expect-error }}
+        @onChange={{field.setValue}}
+        @isDisabled={{@isDisabled}}
+        @isReadOnly={{@isReadOnly}}
+        @rootTestSelector={{@rootTestSelector}}
+        name={{@name}}
+        ...attributes
+      >
+        <:label>{{yield to='label'}}</:label>
+        <:hint>{{yield to='hint'}}</:hint>
+        <:secondary as |secondary|>
+          {{yield
+            (hash
+              CharacterCount=(component
+                (ensure-safe-component secondary.CharacterCount)
+              )
+            )
+            to='secondary'
+          }}
+        </:secondary>
+      </Form::Fields::Textarea>
+    {{else if (this.hasLabelArgAndHintBlock @label (has-block 'hint'))}}
+      <Form::Fields::Textarea
+        @label={{@label}}
+        @error={{this.mapErrors field.rawErrors}}
+        @value={{this.assertString field.value}}
+        {{! @glint-expect-error }}
+        @onChange={{field.setValue}}
+        @isDisabled={{@isDisabled}}
+        @isReadOnly={{@isReadOnly}}
+        @rootTestSelector={{@rootTestSelector}}
+        name={{@name}}
+        ...attributes
+      >
+        <:hint>{{yield to='hint'}}</:hint>
+        <:secondary as |secondary|>
+          {{yield
+            (hash
+              CharacterCount=(component
+                (ensure-safe-component secondary.CharacterCount)
+              )
+            )
+            to='secondary'
+          }}
+        </:secondary>
+      </Form::Fields::Textarea>
+    {{else}}
+      {{! Argument-only case }}
+      <Form::Fields::Textarea
+        @label={{@label}}
+        @hint={{@hint}}
+        @error={{this.mapErrors field.rawErrors}}
+        @value={{this.assertString field.value}}
+        {{! @glint-expect-error }}
+        @onChange={{field.setValue}}
+        @isDisabled={{@isDisabled}}
+        @isReadOnly={{@isReadOnly}}
+        @rootTestSelector={{@rootTestSelector}}
+        name={{@name}}
+        ...attributes
+      >
+        <:secondary as |secondary|>
```

Comment:
Same thing here:

Does this need a `(has-block 'secondary')` wrapped around it?